build-tools/scraper.js

```diff
@@ -0,0 +1,301 @@
+                };
+
+                type = 'List<' + type + '>';
+
+                anonymousTypesToGenerate.push(typeDef);
+
+            }
+        }
+
+        return type;
+    } else {
+        var typeDefinition = findTypeDefinition(resolveName(prop.$ref));
+
+        if (!isPrimitiveOrArray(typeDefinition.type)) {
+            if (typeDefinition.type != currentType) {
+                addDependencyIfNotGenerated(prop.$ref);
+            }
+            return prop.$ref;
+        } else {
+            var type = typeDefinition.type;
+            var resolvedType = primitiveObjectTypes[type] || type;
+
+            if (prop.optional) {
+                if (primitiveOptionalObjectTypes.hasOwnProperty(type)) {
+                    resolvedType = primitiveOptionalObjectTypes[type];
+                }
+            }
+
+            if (type == 'array') {
+                addDependencyIfNotGenerated(typeDefinition.items.$ref);
+                return 'List<' + typeDefinition.items.$ref + '>';
+            }
+
+            return resolvedType;
+        }
+
+    }
+}
+
+function generateJavaClassForType(typeDefinition) {
+    var result = 'public static class ' + typeDefinition.id + ' {';
+
+    if (typeDefinition.properties == undefined) {
+        debugger;
+    }
+
+    typeDefinition.properties.forEach(function (prop) {
+        result += '\r\n\t@JsonProperty';
+
+        if (!prop.optional) {
+            result += '(required = true)';
+        }
+
+        result += '\r\n';
+
+        result += '\tpublic ' + generateJavaTypeEquivalent(typeDefinition.id, prop) + ' ' + prop.name + ';\r\n';
+    });
+
+    result += '}';
+
+    return result;
+}
+
+function addDependencyIfNotGenerated(dependencyString) {
+    var resolved = resolveName(dependencyString);
+
+    if (!dependencyExists(resolved) && !isPrimitiveOrArray(findTypeDefinition(resolved).type)) {
+        dependencies.push(resolved);
+    }
+}
+
+function dependencyExists(resolvedDependency) {
+    return dependencies.filter(function (existing) {
+            return resolvedDependency.name == existing.name && resolvedDependency.domain == existing.domain;
+        }).length != 0 ||
+        alreadyGenerated.filter(function (existing) {
+            return resolvedDependency.name == existing.name && resolvedDependency.domain == existing.domain;
+        }).length != 0;
+}
+
+function generateCommand(commandDef) {
+    var className = commandDef.name.charAt(0).toUpperCase() + commandDef.name.slice(1);
+
+    var hasParams = !!commandDef.parameters;
+    var hasReturns = !!commandDef.returns;
+
+    var paramsTypeName = className + 'Request';
+    var returnsTypeName = className + 'Response';
+
+    var result = '' +
+        '@ChromeDevtoolsMethod\r\n' +
+        'public JsonRpcResult ' + commandDef.name + '(JsonRpcPeer peer, JSONObject params) {\r\n';
+
+    if (hasParams) {
+        result += '' +
+            '\tfinal ' + paramsTypeName + ' = mObjectMapper.convertValue\r\n' +
```

Comment:
what happens with all these tab characters? does IntelliJ convert them to spaces or what?